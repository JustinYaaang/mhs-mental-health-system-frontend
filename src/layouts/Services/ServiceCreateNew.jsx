import React, { Component } from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// @material-ui/core components
import withStyles from '@material-ui/core/styles/withStyles'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import TrustServiceForm from 'views/Forms/TrustServiceForm.jsx'

const styles = {
  cardCategoryWhite: {
    '&,& a,& a:hover,& a:focus': {
      color: 'rgba(255,255,255,.62)',
      margin: '0',
      fontSize: '14px',
      marginTop: '0',
      marginBottom: '0'
    },
    '& a,& a:hover,& a:focus': {
      color: '#FFFFFF'
    }
  },
  cardTitleWhite: {
    color: '#FFFFFF',
    marginTop: '0px',
    minHeight: 'auto',
    fontWeight: '300',
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: '3px',
    textDecoration: 'none',
    '& small': {
      color: '#777',
      fontSize: '65%',
      fontWeight: '400',
      lineHeight: '1'
    }
  }
}
/**
 * Component for creating a new trust
 */
class ServiceCreateNew extends Component {
  constructor (props) {
    super(props)
    console.log(this.props.history)
  }

  componentWillMount () {

  }

  render () {
    return (
      <GridContainer>
        <GridItem xs={12} sm={12} md={12}>
          <TrustServiceForm hasDetails={false} organization={'service'} history={this.props.history}/>
        </GridItem>
      </GridContainer>
    )
  }
}

ServiceCreateNew.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(ServiceCreateNew)
