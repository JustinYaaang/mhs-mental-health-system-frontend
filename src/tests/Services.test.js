import React from 'react'
import { shallow,mount } from 'enzyme'
import ServiceDetails from '../layouts/Services/ServiceDetails'
import ServiceList from '../layouts/Services/ServiceList'
import ServiceCreateManager from '../layouts/Services/ServiceCreateManager'
import ServiceCreateNew from '../layouts/Services/ServiceCreateNew'
import ServiceUserDetails from '../layouts/Services/ServiceUserDetails'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('Service Details', () => {
  it('should render correctly', async () => {
    const component = mount(<ServiceDetails />)

    expect(component).toMatchSnapshot()
  })
})
describe('Service List', () => {
  it('should render correctly', async () => {
    const component = mount(<ServiceList />)

    expect(component).toMatchSnapshot()
  })
})

describe('Service Create Manager', () => {
  it('should render correctly', async () => {
    const component = mount(<ServiceCreateManager />)

    expect(component).toMatchSnapshot()
  })
})
describe('Service Create New', () => {
  it('should render correctly', async () => {
    const component = mount(<ServiceCreateNew />)

    expect(component).toMatchSnapshot()
  })
})
describe('Service User Details', () => {
  it('should render correctly', async () => {
    const component = mount(<ServiceUserDetails />)

    expect(component).toMatchSnapshot()
  })
})
