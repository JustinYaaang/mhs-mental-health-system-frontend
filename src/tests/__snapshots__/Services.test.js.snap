// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Create Manager should render correctly 1`] = `
<ServiceCreateManager
  classes={
    Object {
      "cardCategoryWhite": "ServiceCreateManager-cardCategoryWhite-5",
      "cardTitleWhite": "ServiceCreateManager-cardTitleWhite-6",
    }
  }
/>
`;

exports[`Service Create New should render correctly 1`] = `
<ServiceCreateNew
  classes={
    Object {
      "cardCategoryWhite": "ServiceCreateNew-cardCategoryWhite-7",
      "cardTitleWhite": "ServiceCreateNew-cardTitleWhite-8",
    }
  }
/>
`;

exports[`Service Details should render correctly 1`] = `
<ServiceDetails
  classes={
    Object {
      "cardCategoryWhite": "ServiceDetails-cardCategoryWhite-1",
      "cardTitleWhite": "ServiceDetails-cardTitleWhite-2",
    }
  }
/>
`;

exports[`Service List should render correctly 1`] = `
<ServiceList
  classes={
    Object {
      "cardCategoryWhite": "ServiceList-cardCategoryWhite-3",
      "cardTitleWhite": "ServiceList-cardTitleWhite-4",
    }
  }
/>
`;

exports[`Service User Details should render correctly 1`] = `
<ServiceUserDetails
  classes={
    Object {
      "cardCategoryWhite": "ServiceUserDetails-cardCategoryWhite-9",
      "cardTitleWhite": "ServiceUserDetails-cardTitleWhite-10",
    }
  }
/>
`;
